storybook/src/components/TextInput/TextInput.docs.mdx

```diff
@@ -14,14 +14,116 @@ import README from "../../../../packages/css/src/components/text-input/README.md
 
 ## Examples
 
+### Type: Password
+
+Creates an input where the user can enter a password.
+The characters entered are hidden, usually represented by dots or asterisks, so that the actual text is not visible.
+
+Use the password type when the input requires sensitive information, like passwords or PINs.
+It ensures that the input is not readable by others who might be looking at the screen.
+
+Guidelines:
+
+1. Allow the user’s password manager to automatically enter the password through `autocomplete="current-password"`.
+   When asking for a new password, use `autocomplete="new-password"` instead.
+2. Add a `minlength` attribute to ensure passwords meet a minimum length requirement.
+3. Use the `pattern` attribute to enforce password policies, like including numbers and special characters.
+4. If the password is a numeric PIN, add `inputmode="numeric"`.
+   Devices with virtual keyboards then switch to a numeric keypad layout which makes entering the password easier.
+
+<Canvas of={TextInputStories.Password} />
+
+### Type: Email address
+
+This field helps the user enter an email address.
+It has built-in validation to check for a valid email format.
+However, do not rely only on the browser’s validation.
+
+The `email` input field looks like a standard text input.
+On some devices, it may show an email-specific keyboard.
+
+Guidelines:
+
+1. Set `autocomplete="email"` to help browsers autofill the user’s email address.
+2. Use the `required` attribute to make sure the field is filled before the form is submitted.
+
+<Canvas of={TextInputStories.EmailAddress} />
+
+### Type: Web address
+
+This field helps the user enter an web address or URL.
+It has built-in validation to check for a valid format.
+However, do not rely only on the browser’s validation.
+
+The `url` input field looks like a standard text input.
+On some devices, it may show a URL-specific keyboard to aid in entering web addresses.
+
```

Comment:
We should add that you should '[use autocapitalize="none", autocorrect="off" and spellcheck="false" to stop browsers automatically changing user input on fields that expect grammatically incorrect data—like email addresses and passwords](https://adamsilver.io/blog/form-design-from-zero-to-hero-all-in-one-blog-post/)'.

Reply:
Checkydycheck!